### ISSUE TITLE
Order of UDT definitions should not matter

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#190]
**Describe the bug**
Some weird values show up in the Locals window when debugging a function defined as a User Defined Type.

**To Reproduce**
Run ```Main```:
```VBA
Module Test
    Public Enum ENU
        x = 1
        y = 2
    End Enum

    Public Type UDT
        e As ENU
        u2 As UDT2
    End Type
    
    Public Type UDT2
        v As Variant
        b As Boolean
    End Type

    Public Sub Main()
        Dim u As UDT
        
        InspectUDT
    End Sub
    
    Public Function InspectUDT() As UDT
        Stop
    End Function
End Module
```
and when the ```Stop``` line is hit go and expand the function in the Locals window.

**Screenshots**
![test](https://i.stack.imgur.com/WV696.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.10.975

**Additional Context**
Same issue in previous version of tB.

**Comment by WaynePhillipsEA** [comment]
This turned out to be a more general problem of UDT handling.  If you move the definition of Type UDT2 above Type UDT, then the problem is avoided.

**Comment by Kr00l** [comment]
Will this "limitation" be removed that the UDT, Const, etc. need to be in order?
I kinda like that "Order", since on procedure level you cannot declare a variable at the end but use it in the beginning..

**Comment by cristianbuse** [comment]
In VBA, the above code is actually not allowed. It gives a "Forward reference to user-defined type" compile error. So, yes, it forces the order.

**Comment by cristianbuse** [comment]
I think @Kr00l raised a very interesting point as the order is restricted in VBA at compile time as I mentioned above. However, the following code has the UDT2 definition above the UDT definition but still weird stuff is showing in the Locals window:  
```VBA
Module Test
    Public Enum ENU
        x = 1
        y = 2
    End Enum

    Private Type UDT2
        v As Variant
        b As Boolean
    End Type

    Public Type UDT
        e As ENU
        u2 As UDT2
    End Type

    Public Sub Main()
        Dim u As UDT
        
        u = CreateUDT(x, Null)
        Stop
    End Sub

    Public Function CreateUDT(ByVal e As ENU, ByVal v As Variant) As UDT
        CreateUDT.e = e
        With CreateUDT.u2
            If IsObject(v) Then 
                Set .v = v
            Else
                .v = v
            End If
        End With
    End Function
End Module
```

**Comment by WaynePhillipsEA** [comment]
@cristianbuse that's actually a different issue.  The inner assignment on the inner .v is overwriting the outer .e, which you can see when stepping through the code in CreateUDT.  It looks like the nested UDT accessor on the with-block is not quite working there.

**Comment by cristianbuse** [comment]
@WaynePhillipsEA  Thanks! Good to know.

**Comment by WaynePhillipsEA** [comment]
@cristianbuse the with-block issue mentioned above should be resolved in v0.10.1076 now.

**Comment by WaynePhillipsEA** [comment]
This is now (finally!) fixed in v0.14.10.  Thanks for reporting!
